src/components/Disputes/PrecedenceCampaign/PrecedenceCampaignModal.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react'
+import { Button, GU, Modal, textStyle, useTheme, useViewport } from '@aragon/ui'
+
+import pcModalSvg from './assets/PCModal.svg'
+
+function PrecedenceCampaignModal({ opened, onClose }) {
+  const { width, height } = useViewport()
+  const theme = useTheme()
+
+  const verticalMode = width < 1000
+  const modalHeight = verticalMode
+    ? height - 40
+    : Math.max(700, Math.min(620, height - 40))
+
+  return (
+    <div>
+      <Modal
+        closeButton={false}
+        padding={0}
+        visible={opened}
+        width={Math.min(1055, width - 40)}
```

Comment:
Interesting, I wonder if there's a better way to do this cc @bpierre

Reply:
I think that’s how we generally do it, using `Math.min` / `Math.max` to add constraints to the modal dimensions.

In this specific case we could also use `max-width: 1055px; width: calc(100vw - 40px)` if we wanted to stay in CSS only, but I think that doing it in JS makes sense since we are doing other calculations.

Otherwise I think we could use more `GU` here :-)